app/dashboard/views/alias_log.py

```diff
@@ -1,27 +1,32 @@
-from dataclasses import dataclass
-
-import arrow
-from flask import render_template, flash, redirect, url_for
+from flask import render_template, flash, redirect, url_for, abort
 from flask_login import login_required, current_user
 
 from app.dashboard.base import dashboard_bp
 from app.extensions import db
 from app.models import GenEmail, ForwardEmailLog, ForwardEmail
 
+LIMIT = 15
+
 
-@dataclass
 class AliasLog:
-    website_email: str
-    website_from: str
-    alias: str
-    when: arrow.Arrow
-    is_reply: bool
-    blocked: bool
+    __slots__ = [
+        "website_email",
+        "website_from",
+        "alias",
+        "when",
+        "is_reply",
+        "blocked",
+    ]  # memory efficiency
```

Comment:
Ah could you tell more about why we should not use `dataclass` please?

Reply:
For no good reason actually, but without the dataclass dependent, the app can also run on py36. Other solution is to install dataclass package on pypi for py<37, but I think it's overcomplicated

Reply:
Ah I think we don't have to support python 3.6 as the code is meant to be run inside a container. There are alternatives to dataclasses but they (attrs for ex) but dataclasses has the advantage of being included in the standard lib.

Reply:
Couldn't agree more on the standard lib of py37. I think SL should support as many Python versions as it can, since the lower Python version is, the more mature and the number of developers are working with it. You're right that the code is meant to run inside a container, but for developing the app, for myself, I have to set up another version of Python. So to me, even this is not a big concern, I will create another PR regarding just this story and let it be there, we will consider later.

Reply:
ah ok I see. In this case https://github.com/python-attrs/attrs could be a good option indeed.